t/25-cache-service.t

```diff
@@ -0,0 +1,454 @@
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+
+BEGIN {
+    unshift @INC, 'lib', 't/lib';
+    use Mojo::File qw(path tempdir);
+    use FindBin;
+    use File::Path qw(remove_tree make_path);
+
+    my $basedir = path(tempdir, 't', 'cache.d');
+    remove_tree($basedir);
+    $ENV{CACHE_DIR}      = path($basedir, 'cache');
+    $ENV{OPENQA_BASEDIR} = $basedir;
+    $ENV{OPENQA_CONFIG}  = path($basedir, 'config')->make_path;
+    path($ENV{OPENQA_CONFIG})->child("workers.ini")->spurt("
+[global]
+CACHEDIRECTORY = $cachedir
+CACHELIMIT = 100");
+}
+
+use strict;
+use warnings;
+# Avoid warning error: Name "Config::IniFiles::t/cache.d/cache/config/workers.ini" used only once
+use OpenQA::Worker::Cache;
+OpenQA::Worker::Cache->from_worker;
+use Test::More;
+use Test::Warnings;
+use OpenQA::Utils;
+use File::Spec::Functions qw(catdir catfile);
+use Cwd qw(abs_path getcwd);
+use Mojolicious;
+use IO::Socket::INET;
+use Mojo::Server::Daemon;
+use Mojo::IOLoop::Server;
+use POSIX '_exit';
+use Mojo::IOLoop::ReadWriteProcess qw(queue process);
+use Mojo::IOLoop::ReadWriteProcess::Session 'session';
+use OpenQA::Test::Utils qw(fake_asset_server cache_minion_worker cache_worker_service);
+use Mojo::Util qw(md5_sum);
+use constant DEBUG => $ENV{DEBUG} // 0;
+use OpenQA::Worker::Cache::Request;
+
+my $sql;
+my $sth;
+my $result;
+my $dbh;
+my $filename;
+my $serverpid;
+my $openqalogs;
+my $cachedir = $ENV{CACHE_DIR};
+
+my $db_file = "$cachedir/cache.sqlite";
+$ENV{LOGDIR} = catdir(getcwd(), 't', 'cache.d', 'logs');
+my $logfile = catdir($ENV{LOGDIR}, 'cache.log');
+my $port    = Mojo::IOLoop::Server->generate_port;
+my $host    = "http://localhost:$port";
+
+make_path($ENV{LOGDIR});
+use OpenQA::Worker::Cache::Client;
+
+my $cache_client = OpenQA::Worker::Cache::Client->new();
+
+# reassign STDOUT, STDERR
+sub _port { IO::Socket::INET->new(PeerAddr => '127.0.0.1', PeerPort => shift) }
+
+END { session->clean }
+
+my $daemon;
+my $cache_service        = cache_worker_service;
+my $worker_cache_service = cache_minion_worker;
+
+my $server_instance = process sub {
+    # Connect application with web server and start accepting connections
+    $daemon = Mojo::Server::Daemon->new(app => fake_asset_server, listen => [$host])->silent(!DEBUG);
+    $daemon->run;
+    _exit(0);
+};
+
+sub start_server {
+    $server_instance->set_pipes(0)->separate_err(0)->blocking_stop(1)->channels(0)->restart;
+    $cache_service->set_pipes(0)->separate_err(0)->blocking_stop(1)->channels(0)->restart->restart;
+    $worker_cache_service->restart;
+    sleep 2 and diag "Wait server to be available" until $cache_client->available;
+    return;
+}
+
+sub test_default_usage {
+    my ($id, $a) = @_;
+    my $asset_request = $cache_client->request->asset(id => $id, asset => $a, type => 'hdd', host => $host);
+
+    if ($asset_request->enqueue) {
+        1 until $asset_request->processed;
```

Comment:
Is busy waiting really necessary here?

Reply:
Argh, that's a leftover, those calls do not need to be busy at all